fields.go

```diff
@@ -175,11 +176,18 @@ func (mf *mysqlField) scanType() reflect.Type {
 		}
 		return scanTypeNullFloat
 
-	case fieldTypeDecimal, fieldTypeNewDecimal, fieldTypeVarChar,
-		fieldTypeBit, fieldTypeEnum, fieldTypeSet, fieldTypeTinyBLOB,
-		fieldTypeMediumBLOB, fieldTypeLongBLOB, fieldTypeBLOB,
-		fieldTypeVarString, fieldTypeString, fieldTypeGeometry, fieldTypeJSON,
-		fieldTypeTime:
+	case fieldTypeVarChar, fieldTypeString, fieldTypeVarString:
+		return scanTypeString
+
+	case fieldTypeTinyBLOB, fieldTypeMediumBLOB, fieldTypeLongBLOB, fieldTypeBLOB:
+		if mf.charSet == collations[defaultCollation] {
+			return scanTypeString
+		}
+		fallthrough
```

Comment:
See https://dev.mysql.com/doc/refman/8.0/en/c-api-data-structures.html

> To distinguish between binary and nonbinary data for string data types, check whether the charsetnr value is 63.

So correct code is this:

```go
// charsetnr == 63 means this column is binary.
// https://dev.mysql.com/doc/refman/8.0/en/c-api-data-structures.html
if mf.charSet == 63 {
    return scanTypeRawBytes
}
return scanTypeString
```